### PR TITLE
docs(CONTRIBUTING): add GitHub Issue and PR template

### PR DESCRIPTION
**Pull request by staltz**
Also about checklist to include new operators

Fixes issues #1355 and #1429

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<!--
```
Comment (kwonoj) [inline .github/ISSUE_TEMPLATE.md]: is this comment appears up in template? view markdown in PR shows only non commented parts.

**Comment by todoubaba** [comment]
I prefer not use the Issue and PR template, look at the angular2's new issures, it's annoying and not provide clearer intents.

**Comment by kwonoj** [comment]
I'm still in favor of (at least trying) template to see if this can help anyone who want to contribute can leverage it. After rolling this out and notice if this confuses people, can be easily taken out.

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+Thank you for raising your concerns, we appreciate your feedback and contributions to this repository.
+
+Before you continue, consider the following:
+
+If you have a "How do I do ...?" question, it is better for you and for us that this question is placed in [StackOverflow](http://stackoverflow.com/questions/tagged/rxjs) or some chat channel. This way, you are making it easier for others to learn from your experiences too.
```
Comment (blesh) [inline .github/ISSUE_TEMPLATE.md]: I think StackOverflow actually has an rxjs5 tag.

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+These "Issues" are meant only for technical problems, bugs, and proposals related to the library.
+
+If your issue is a bug, please follow the format below:
+-->
+
+**What version of RxJS I am using:**
```
Comment (blesh) [inline .github/ISSUE_TEMPLATE.md]: I'd like this to be a little more concise:

```
**RxJS version:**

**Code to reproduce:**

**Expected behavior:**

**Actual behavior:**

**Additional information:**
```

File: doc/operator-creation.md [hunk 4]
```diff
@@ -128,7 +128,20 @@ There are a few things to know and (try to) understand while developing operator
...
-   
+
+### Submitting a PR for an Operator
+
+Please complete these steps for each new operator added to RxJS as a pull request:
+
+- Add the operator to either Core or KitchenSink
+- It should have a `-spec.ts` tests file covering the canonical corner cases, with marble diagram tests
+- It should have a `asDiagram` test case too, if possible
+- It Should have a type definition test at the end of the spec to verify type definition for various use cases
```
Comment (blesh) [inline doc/operator-creation.md]: It **s**hould

Comment (blesh) [inline doc/operator-creation.md]: Could we add an explanation of what a "Type Definition Test" is, and link it from here?

File: doc/operator-creation.md [hunk 5]
```diff
@@ -128,7 +128,20 @@ There are a few things to know and (try to) understand while developing operator
    the shared underlying subscription.
    - When you `subscribe` to an Observable, the functions or Observer you passed are used to create the final `destination`
    `Subscriber` for the chain. It's this `Subscriber` that is really also the shared `Subscriptoin` for the operator chain.
-   
+
+### Submitting a PR for an Operator
+
+Please complete these steps for each new operator added to RxJS as a pull request:
```
Comment (blesh) [inline doc/operator-creation.md]: Perhaps we could change the language here to be a little more explicit. Rather that "it should, it should", etc. We could use "It MUST including" and "Consider including". "Should" is a little ambiguous between "this is optional" and "this is not optional".

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (staltz) [inline .github/ISSUE_TEMPLATE.md]: Yup it appears. I saw it in Typescript repo's issue template.

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (blesh) [inline .github/ISSUE_TEMPLATE.md]: I'd like to reword this section, honestly. (Sorry, haha)

```
QUESTIONS: Please direct questions to [StackOverflow](http://stackoverflow.com/questions/tagged/rxjs), our maintainers and the larger community actively answer questions there, and it's a much better format for Q&A. It's also more likely to help others with similar questions.
```

File: doc/operator-creation.md (hunk 4 above)
Comment (staltz) [inline doc/operator-creation.md]: @kwonoj can you help explain what a Type Definition test is and where to find a good example?

**Comment by staltz** [comment]
@kwonoj can you help explain what a Type Definition test is and where to find a good example?

**Comment by staltz** [comment]
PS @blesh I updated the PR

**Comment by kwonoj** [comment]
Type definition test can be found at opened PR https://github.com/ReactiveX/RxJS/pull/1189 / https://github.com/ReactiveX/RxJS/pull/1414 (it's not checked in yet) .

Basically it tries to test interface of operators to conform

- operator input signature accepts parameter as expected
- operator chain, or subscription correctly delivers context of value type `<T>`

**Comment by kwonoj** [comment]
Change looks good to me, explanation regarding type definition test can be added later in detail once PR's checked in.

**Comment by kwonoj** [comment]
Merged with e92a7b3, thanks @staltz

**Comment by staltz** [comment]
> Change looks good to me, explanation regarding type definition test can be added later in detail once PR's checked in.

Yeah you can do that since you understand better about those tests.
